designer/server/src/main/scala/pl/touk/nussknacker/ui/server/AkkaHttpBasedRouteProvider.scala

```diff
@@ -165,7 +165,7 @@ class AkkaHttpBasedRouteProvider(
         val validator = new UIProcessValidator(
           processingTypeData.name,
           ProcessValidator.default(processingTypeData.designerModelData.modelData),
-          processingTypeData.deploymentData.scenarioPropertiesConfig,
+          processingTypeData.deploymentData.scenarioPropertiesConfig.scenarioPropertiesConfig,
```

Comment:
1. It looks weird that we have some snake with doubled names `.scenarioPropertiesConfig.scenarioPropertiesConfig`
2. Both validator and newProcessPreparer don't need to know about properties URL. They need properties to:
a) validator - do validations against properties validators
b) newProcessPreparer - prepare scenario with default values that are available in the scenario properties definition
Let's avoid passing things that are not needed to classes

Reply:
1. That is basically why I changed scenarioPropertiesConfig to propertiesConfig in the model.conf file. Currently i am at a loss because this is kinda ridiculous that we have scenarioPropertiesConfig.scenarioPropertiesConfig but I can't really come up with a new name since it would be nice if the domain objects had the same name as in the config, to avoid confusion. Rename `DeploymentData.scenarioPropertiesConfig` to `DeploymentData.scenarioProperties`? wdyt?

2. We're only passing `
          processingTypeData.deploymentData.scenarioPropertiesConfig.scenarioPropertiesConfig` which doesn't contain the url, I dont understand the issue.

Reply:
I think that this solution would be much easier if we moved this URL configuration outside of `scenarioType` config.

Do we have a requirement that this URL should be different for each processing type?
If not, we can move this URL on top level and remove it from `DeploymentData` and all other places. This change would reduce to ~20 LOC

`DeploymentData` is designed to keep objects that are derivates of scenarioType > deploymentConfig

Reply:
The curent requirement is to have the url configurable for every processing mode so Batch, Streaming, etc. Does this make your idea possible?

Reply:
You mean different url for batch different for streaming?

Reply:
Yes, the properties are different for these processing modes so it's natural that we have different properties docs for these.

Reply:
Ok, so it have to be inside `scenarioType` config but I would put it into separate field in the `DeploymentData` - the same as we have in the config. Thanks to that, we won't need to think how to avoid `scenarioPropertiesConfig.scenarioPropertiesConfig` repetition